diff -ur sarab.orig/etc/sarab.conf sarab/etc/sarab.conf
--- sarab.orig/etc/sarab.conf	2004-09-08 23:05:59.000000000 -0500
+++ sarab/etc/sarab.conf	2005-07-27 12:36:09.000000000 -0500
@@ -55,6 +55,12 @@
 # Default="no"
 SARAB_VERBOSE="no"
 
+# If non-empty, DAR_ENCRYPTION_OPTIONS contains the cipher options and key to be
+# used to encrypt the backups.  See the dar(1) for a description of what is
+# possible.  
+# eg. DAR_ENCRYPTION_OPTIONS="--crypto-block 20480 --key blowfish:My_CompleX_key_123"
+DAR_ENCRYPTION_OPTIONS=""
+
 ##################################################################
 # DO NOT EDIT BELOW THIS LINE UNLESS YOU KNOW WHAT YOU ARE DOING #
 ##################################################################
Only in sarab/etc: sarab.conf~
diff -ur sarab.orig/etc/sarab.dcf sarab/etc/sarab.dcf
--- sarab.orig/etc/sarab.dcf	2004-09-08 23:05:59.000000000 -0500
+++ sarab/etc/sarab.dcf	2005-07-27 12:23:22.000000000 -0500
@@ -113,16 +113,6 @@
 --exclude-compression "*.Z"
 
 
-# --key <string>
-# This option will scramble the archive using <string> as the pass-phrase.
-# A scrambled archive can only be read if the same pass-phrase is given.
-# This should not be considered a secure solution, that is why we say 
-# "scramble" instead of "encrypt". Still, this option can prevent 
-# unexperienced people from looking at your data.
-# Example: (But create your own key!)
-# --key "My_CompleX_key_ADDS_some_Security-Easy_as_123"
-
-
 # Many of the more complex and advanced options of Dar are not listed above.
 # For experienced users, all valid Dar options may be used in this file,
 # except what is noted below.
@@ -131,3 +121,7 @@
 # NOTE: Do NOT use the options "--create" and "--ref", because SaraB  
 # automatically generates these during run-time.  If you mistakenly add
 # them to this file, Dar generate an error about duplicate arguments.
+
+# Local Variables:
+# mode: conf
+# End:
Only in sarab/etc: sarab.dcf~
diff -ur sarab.orig/sarab.sh sarab/sarab.sh
--- sarab.orig/sarab.sh	2004-09-08 23:06:16.000000000 -0500
+++ sarab/sarab.sh	2005-07-27 12:37:17.000000000 -0500
@@ -211,7 +211,7 @@
 echo -n "Creating backup with DAR..."
 verbose
 verbose "$DAR_COMMAND $REFERENCE_BASENAME"
-$DAR_COMMAND $REFERENCE_BASENAME
+$DAR_COMMAND $REFERENCE_BASENAME $DAR_ENCRYPTION_OPTIONS
 if [ "$?" != "0" ]; then
    echo "ERROR: Error when executing the backup with DAR. The attempted command was... "
    echo "$DAR_COMMAND $REFERENCE_BASENAME"
@@ -224,7 +224,7 @@
    echo -n "Testing the archive for errors..."
    verbose
    verbose "$DAR_BINARY -t $DESTINATION/$WORK_DIR/$BASENAME --noconf"
-   $DAR_BINARY -t $DESTINATION/$WORK_DIR/$BASENAME --noconf
+   $DAR_BINARY -t $DESTINATION/$WORK_DIR/$BASENAME $DAR_ENCRYPTION_OPTIONS --noconf
    if [ "$?" != "0" ]; then
       echo "ERROR: Error when testing the archive. The attempted command was... "
       echo "$DAR_BINARY -t $DESTINATION/$WORK_DIR/$BASENAME --noconf"
Only in sarab: sarab.sh~
