diff -Naur sarab.orig/sarab.sh sarab/sarab.sh
--- sarab.orig/sarab.sh	2006-01-04 09:45:36.000000000 +0100
+++ sarab/sarab.sh	2006-01-04 09:49:43.000000000 +0100
@@ -34,7 +34,7 @@
 function rotate 
 {
     lines=$(cat $SARAB_ETC/$ROTATION_SCHEDULE | wc -l)
-    firstline="$(head -n 1 $SARAB_ETC/rotation.schedule)"
+    firstline="$(head -n 1 $SARAB_ETC/$ROTATION_SCHEDULE)"
 
     # Copy all but the first line back into rotation schedule
     tail -n $(expr $lines - 1) $SARAB_ETC/$ROTATION_SCHEDULE > $SARAB_ETC/rotation.schedule.temp
