--- sarab.orig/sarab.sh	2004-09-08 23:06:16.000000000 -0500
+++ sarab/sarab.sh	2006-03-01 15:00:31.000000000 -0600
@@ -176,7 +176,7 @@
    REFERENCE_ARCHIVE=$(echo $CURRENT_LINE | cut -f 2 -d" ")
    # Test to see if the reference archive actually exists
    if [ -d "$DESTINATION/$REFERENCE_ARCHIVE/" ]; then     # The reference archive exists  
-      REFERENCE_BASENAME="--ref $(/bin/ls $DESTINATION/$REFERENCE_ARCHIVE/*.dar | head -n 1 | cut -f 1 -d".")"
+      REFERENCE_BASENAME="--ref $(/bin/ls $DESTINATION/$REFERENCE_ARCHIVE/*.dar | head -n 1 | sed -re 's,\.[0-9]+\.dar,,g')"
       # Record information about the reference archive to include in the current archive
       echo "The reference archive for this backup was:" > $DESTINATION/$WORK_DIR/reference_archive.txt
       echo "$(ls -ltr $DESTINATION/$REFERENCE_ARCHIVE/*.dar)" >> $DESTINATION/$WORK_DIR/reference_archive.txt
