diff -ur sarab.orig/etc/sarab.conf sarab/etc/sarab.conf
--- sarab.orig/etc/sarab.conf	2005-07-27 12:36:09.000000000 -0500
+++ sarab/etc/sarab.conf	2005-07-27 12:43:52.000000000 -0500
@@ -56,8 +56,8 @@
 SARAB_VERBOSE="no"
 
 # If non-empty, DAR_ENCRYPTION_OPTIONS contains the cipher options and key to be
-# used to encrypt the backups.  See the dar(1) for a description of what is
-# possible.  
+# used to encrypt the backups.  See the dar(1) manual for a description of what
+# is possible.
 # eg. DAR_ENCRYPTION_OPTIONS="--crypto-block 20480 --key blowfish:My_CompleX_key_123"
 DAR_ENCRYPTION_OPTIONS=""
 
@@ -74,10 +74,10 @@
 BASENAME="$(date +'%m-%d-%Y_%H%M')"
 
 # The location of the DAR executable file
-DAR_BINARY="/usr/local/bin/dar"
+DAR_BINARY="/usr/bin/dar"
 
 # The location of the statically-compiled DAR executable file
-DAR_STATIC="/usr/local/bin/dar_static"
+DAR_STATIC="/usr/bin/dar_static"
 
 # The temporary directory name that backups will be created in. This will be created underneath $DESTINATION.
 WORK_DIR=".sarab_temp_workdir"
@@ -90,4 +90,4 @@
 DAR_COMMAND="$DAR_BINARY --batch $SARAB_ETC/$SARAB_DCF -c $DAR_CREATE --noconf"
 
 # Location and name of temporary log file
-TEMP_LOG="$SARAB_ETC/sarab_temp.log"
+TEMP_LOG="/var/log/sarab_temp.log"
