--- backup-manager-0.7.4/lib/upload-methods.sh	2006-06-20 17:39:16.000000000 +0200
+++ backup-manager-0.7.4-modified/lib/upload-methods.sh	2006-08-07 17:36:14.000000000 +0200
@@ -60,7 +60,7 @@
 
     # Call to backup-manager-upload
     logfile="$(mktemp /tmp/bmu-log.XXXXXX)"
-    su $BM_UPLOAD_SSH_USER -s /bin/sh -c \
+    su $BM_UPLOAD_SSH_USER -c \
     "$bmu $v_switch \
           $k_switch \
           $ssh_purge_switch \
@@ -100,8 +100,8 @@
 
     # Call to backup-manager-upload
     logfile="$(mktemp /tmp/bmu-log.XXXXXX)"
-    su $BM_UPLOAD_SSH_USER -s /bin/sh -command="\
-    $bmu $v_switch \
+    su $BM_UPLOAD_SSH_USER -c \
+    "$bmu $v_switch \
          $k_switch \
          -m=\"ssh-gpg\" \
          -h=\"$bm_upload_hosts\" \
