*** backup-manager-0.7.2/lib/upload-methods.sh	2006-03-29 15:43:18.929295040 +0300
--- backup-manager-0.7.2-modified/lib/upload-methods.sh	2006-03-29 15:43:18.933294432 +0300
***************
*** 54,60 ****
      fi
  
      # Call to backup-manager-upload
!     su $BM_UPLOAD_SSH_USER -s /bin/sh -c \
      "$bmu $v_switch \
            $k_switch \
            -m=\"scp\" \
--- 54,60 ----
      fi
  
      # Call to backup-manager-upload
!     su $BM_UPLOAD_SSH_USER -c \
      "$bmu $v_switch \
            $k_switch \
            -m=\"scp\" \
