diff -Nuar amanda-2.5.2p1.orig/changer-src/chg-multi.sh.in amanda-2.5.2p1/changer-src/chg-multi.sh.in
--- amanda-2.5.2p1.orig/changer-src/chg-multi.sh.in	2007-05-17 10:38:34.000000000 -0700
+++ amanda-2.5.2p1/changer-src/chg-multi.sh.in	2008-06-25 14:57:04.000000000 -0700
@@ -325,7 +325,7 @@
 		echo $answer
 		exit 2
 	    fi
-	    echo `_ '     -> offline'` "$device"` >> $logfile
+	    echo `_ '     -> offline'` "$device" >> $logfile
 	    $MT $MTF $device offline >> $logfile 2>&1
 	    if [ $? -ne 0 ]; then
 		answer=`_ '%s %s: %s: unable to change to slot %s' "$newslot" "$pname" "$device" "$curslot"`
