--- amanda-2.5.2_p1.orig/tape-src/Makefile.am	2007-05-04 11:39:08.000000000 +0000
+++ amanda-2.5.2_p1/tape-src/Makefile.am	2008-03-10 19:06:28.000000000 +0000
@@ -20,6 +20,8 @@ libamtape_la_SOURCES = 	output-file.c \
 
 libamtape_la_LDFLAGS =  -release $(VERSION)
 
+libamtape_la_LIBADD = ../common-src/libamanda.la
+
 ###
 # Because libamanda includes routines (e.g. regex) provided by some system
 # libraries, and because of the way libtool sets up the command line, we
--- amanda-2.5.2_p1.orig/server-src/Makefile.am	2007-05-04 11:39:04.000000000 +0000
+++ amanda-2.5.2_p1/server-src/Makefile.am	2008-03-10 19:07:58.000000000 +0000
@@ -36,6 +36,8 @@ libamserver_la_SOURCES=	amindex.c	change
 
 libamserver_la_LDFLAGS= -release $(VERSION)
 
+libamserver_la_LIBADD = ../tape-src/libamtape.la
+
 ###
 # Because libamanda includes routines (e.g. regex) provided by some system
 # libraries, and because of the way libtool sets up the command line, we
