diff -u -r --show-c-function --new-file --exclude-from=/home/martinea/src.orig/amanda.diff --ignore-matching-lines='$Id:' amanda-2.5.1p3.new/server-src/amstatus.pl.in amanda-2.5.1p3.new.amstatus/server-src/amstatus.pl.in
--- amanda-2.5.1p3.new/server-src/amstatus.pl.in	2007-02-01 14:17:01.000000000 -0500
+++ amanda-2.5.1p3.new.amstatus/server-src/amstatus.pl.in	2007-02-12 18:35:11.000000000 -0500
@@ -763,9 +763,9 @@ while(<AMDUMP>) {
 	}
 	elsif(/splitting chunk that started at \d*kb after (\d*)kb/) {
 		$ntchunk{$nb_tape}++;
-		$ntsize{$nb_tape} += $1;
-		$ntesize{$nb_tape} += $1;
-		$ntchunk_size += $1;
+		$ntsize{$nb_tape} += $1 / $unitdivisor;
+		$ntesize{$nb_tape} += $1 / $unitdivisor;
+		$ntchunk_size += $1 / $unitdivisor;
 	}
 	elsif(/driver: FINISHED/) {
 		$driver_finished = 1;
Binary files amanda-2.5.1p3.new/server-src/.amstatus.pl.in.swp and amanda-2.5.1p3.new.amstatus/server-src/.amstatus.pl.in.swp differ
