diff -Naur backintime-0.9.26.orig/common/Makefile.template backintime-0.9.26/common/Makefile.template
--- backintime-0.9.26.orig/common/Makefile.template	2010-01-13 09:34:18.000000000 +0100
+++ backintime-0.9.26/common/Makefile.template	2010-01-13 09:34:35.000000000 +0100
@@ -17,10 +17,6 @@
 	install -d $(DEST)/share/backintime/plugins
 	install --mode=644 ../plugins/usercallbackplugin.py $(DEST)/share/backintime/plugins
 	
-	#install copyright file
-	install -d $(DEST)/share/doc/backintime-common
-	install --mode=644 debian_specific/copyright $(DEST)/share/doc/backintime-common
-	
 	#install doc file(s)
 	install -d $(DEST)/share/doc/backintime
 	install --mode=644 ../AUTHORS $(DEST)/share/doc/backintime
diff -Naur backintime-0.9.26.orig/gnome/Makefile.template backintime-0.9.26/gnome/Makefile.template
--- backintime-0.9.26.orig/gnome/Makefile.template	2010-01-13 09:34:17.000000000 +0100
+++ backintime-0.9.26/gnome/Makefile.template	2010-01-13 09:34:46.000000000 +0100
@@ -15,10 +15,6 @@
 	install -d $(DEST)/share/backintime/plugins
 	install --mode=644 ../plugins/gnome*.py $(DEST)/share/backintime/plugins
 	
-	#install copyright file
-	install -d $(DEST)/share/doc/backintime-gnome
-	install --mode=644 ../common/debian_specific/copyright $(DEST)/share/doc/backintime-gnome
-	
 	#install man file(s)
 	install -d $(DEST)/share/man/man1
 	install --mode=644 man/C/*.gz $(DEST)/share/man/man1
diff -Naur backintime-0.9.26.orig/kde4/Makefile.template backintime-0.9.26/kde4/Makefile.template
--- backintime-0.9.26.orig/kde4/Makefile.template	2010-01-13 09:34:18.000000000 +0100
+++ backintime-0.9.26/kde4/Makefile.template	2010-01-13 09:35:08.000000000 +0100
@@ -19,10 +19,6 @@
 	install -d $(DEST)/bin
 	install backintime-kde4 $(DEST)/bin
 	
-	#install copyright file
-	install -d $(DEST)/share/doc/backintime-kde4
-	install --mode=644 ../common/debian_specific/copyright $(DEST)/share/doc/backintime-kde4
-	
 	#install .desktop file(s)
 	install -d $(DEST)/share/applications/kde4
 	install --mode=644 *.desktop $(DEST)/share/applications/kde4
