diff -ruN boxbackup-0.09.orig/bin/bbackupd/bbackupd-config boxbackup-0.09/bin/bbackupd/bbackupd-config
--- boxbackup-0.09.orig/bin/bbackupd/bbackupd-config	2005-04-18 15:11:17.603115864 +0600
+++ boxbackup-0.09/bin/bbackupd/bbackupd-config	2005-04-18 17:39:43.795168872 +0600
@@ -56,7 +56,7 @@
 Usage:
 	bbackupd-config config-dir backup-mode account-num server-hostname working-dir backup-dir [more backup directories]
 
-config-dir usually /etc/box
+config-dir usually /etc/boxbackup
 backup-mode is lazy or snapshot
     lazy mode runs continously, uploading files over a specified age
     snapshot mode uploads a snapshot of the filesystem when instructed explicitly
@@ -92,7 +92,7 @@
 }
 
 # default locations
-my $default_config_location = '/etc/box/bbackupd.conf';
+my $default_config_location = '/etc/boxbackup/bbackupd.conf';
 
 # command line parameters
 my ($config_dir,$backup_mode,$account_num,$server,$working_dir,@tobackup) = @ARGV;
@@ -540,7 +540,7 @@
    more files will be backed up. You want to know about this.
 
 6) Start the backup daemon with the command
-      /usr/local/bin/bbackupd$daemon_args
+      /usr/sbin/bbackupd$daemon_args
    in /etc/rc.local, or your local equivalent.
    Note that bbackupd must run as root.
 __E
@@ -550,7 +550,7 @@
 
 7) Set up a cron job to run whenever you want a snapshot of the
    file system to be taken. Run the command
-      /usr/local/bin/bbackupctl -q$ctl_daemon_args sync
+      /usr/sbin/bbackupctl -q$ctl_daemon_args sync
 __E
 }
 print <<__E;
diff -ruN boxbackup-0.09.orig/bin/bbstored/bbstored-config boxbackup-0.09/bin/bbstored/bbstored-config
--- boxbackup-0.09.orig/bin/bbstored/bbstored-config	2005-04-18 15:11:17.600116320 +0600
+++ boxbackup-0.09/bin/bbstored/bbstored-config	2005-04-18 17:40:00.205674096 +0600
@@ -57,7 +57,7 @@
 Usage:
 	bbstored-config config-dir server-hostname username [raidfile-config]
 
-config-dir usually /etc/box
+config-dir usually /etc/boxbackup
 server-hostname is the hostname used by clients to connect to this server
 username is the user to run the server under
 raidfile-config is optional. Use if you have a non-standard raidfile.conf file.
@@ -83,7 +83,7 @@
 }
 
 # default locations
-my $default_config_location = '/etc/box/bbstored.conf';
+my $default_config_location = '/etc/boxbackup/bbstored.conf';
 
 # command line parameters
 my ($config_dir,$server,$username,$raidfile_config) = @ARGV;
@@ -270,7 +270,7 @@
 4) Create accounts with bbstoreaccounts
 
 5) Start the backup store daemon with the command
-      /usr/local/bin/bbstored$daemon_args
+      /usr/sbin/bbstored$daemon_args
    in /etc/rc.local, or your local equivalent.
 
 ===================================================================
diff -ruN boxbackup-0.09.orig/infrastructure/BoxPlatform.pm boxbackup-0.09/infrastructure/BoxPlatform.pm
--- boxbackup-0.09.orig/infrastructure/BoxPlatform.pm	2005-04-18 15:11:17.581119208 +0600
+++ boxbackup-0.09/infrastructure/BoxPlatform.pm	2005-04-18 16:35:27.793370248 +0600
@@ -78,7 +78,7 @@
 	close VERSION;
 	
 	# where to put the files
-	$install_into_dir = '/usr/local/bin';
+	$install_into_dir = '$1/usr/sbin';
 	
 	# if it's Darwin,
 	if($build_os eq 'Darwin')
diff -ruN boxbackup-0.09.orig/infrastructure/makeparcels.pl boxbackup-0.09/infrastructure/makeparcels.pl
--- boxbackup-0.09.orig/infrastructure/makeparcels.pl	2005-04-18 15:11:17.581119208 +0600
+++ boxbackup-0.09/infrastructure/makeparcels.pl	2005-04-18 17:52:15.451899680 +0600
@@ -139,11 +139,11 @@
 		{
 			my $exeext = ($build_os eq 'CYGWIN')?'.exe':'';
 			print MAKE "\t(cd bin/$name; $make_command $release_flag)\n";
-			print MAKE "\tcp release/bin/$name/$name$exeext $dir\n";
+			print MAKE "\tcp -R release/bin/$name/$name$exeext $dir/\n";
 		}
 		elsif ($type eq 'script')
 		{
-			print MAKE "\tcp $name $dir\n";
+			print MAKE "\tcp $name $dir/\n";
 			# remove path from script name
 			$name =~ m~/([^/]+)\Z~;
 			$name = $1;
@@ -157,13 +157,13 @@
 	chmod 0755,"parcels/scripts/install-$parcel";
 	
 	my $root = parcel_root($parcel);
-	print MAKE "\tcp parcels/scripts/install-$parcel $dir\n";
+	print MAKE "\tcp parcels/scripts/install-$parcel $dir/\n";
 	print MAKE "\t(cd parcels; tar cf - $root | gzip -9 - > $root.tgz )\n";
 	
 	print MAKE "\n";
 	
 	print MAKE "install-$parcel:\n";
-	print MAKE "\t(cd $dir; ./install-$parcel)\n\n";
+	print MAKE "\t(cd $dir; ./install-$parcel \$(DESTDIR) )\n\n";
 }
 
 print MAKE <<__E;
diff -ruN boxbackup-0.09.orig/lib/common/BoxPortsAndFiles.h boxbackup-0.09/lib/common/BoxPortsAndFiles.h
--- boxbackup-0.09.orig/lib/common/BoxPortsAndFiles.h	2005-04-18 15:11:17.345155080 +0600
+++ boxbackup-0.09/lib/common/BoxPortsAndFiles.h	2005-04-18 17:40:43.200137944 +0600
@@ -54,16 +54,16 @@
 
 // Backup store daemon
 #define BOX_PORT_BBSTORED					(BOX_PORT_BASE+1)
-#define BOX_FILE_BBSTORED_DEFAULT_CONFIG	"/etc/box/bbstored.conf"
+#define BOX_FILE_BBSTORED_DEFAULT_CONFIG	"/etc/boxbackup/bbstored.conf"
 // directory within the RAIDFILE root for the backup store daemon
 #define BOX_RAIDFILE_ROOT_BBSTORED			"backup"
 
 // Backup client daemon
-#define BOX_FILE_BBACKUPD_DEFAULT_CONFIG	"/etc/box/bbackupd.conf"
+#define BOX_FILE_BBACKUPD_DEFAULT_CONFIG	"/etc/boxbackup/bbackupd.conf"
 
 
 // RaidFile conf location efault
-#define BOX_FILE_RAIDFILE_DEFAULT_CONFIG	"/etc/box/raidfile.conf"
+#define BOX_FILE_RAIDFILE_DEFAULT_CONFIG	"/etc/boxbackup/raidfile.conf"
 
 
 #endif // BOXPORTSANDFILES__H
diff -ruN boxbackup-0.09.orig/lib/raidfile/raidfile-config boxbackup-0.09/lib/raidfile/raidfile-config
--- boxbackup-0.09.orig/lib/raidfile/raidfile-config	2005-04-18 15:11:17.450139120 +0600
+++ boxbackup-0.09/lib/raidfile/raidfile-config	2005-04-18 17:40:30.723034752 +0600
@@ -57,7 +57,7 @@
 Usage:
 	raidfile-config config-dir block-size dir0 [dir1 dir2]
 
-config-dir usually /etc/box
+config-dir usually /etc/boxbackup
 block-size must be a power of two, and usually the block or fragment size of your filing system
 dir0, dir1, dir2 are the directories used as the root of the raid file system
 
diff -ruN boxbackup-0.09.orig/lib/raidfile/RaidFileController.h boxbackup-0.09/lib/raidfile/RaidFileController.h
--- boxbackup-0.09.orig/lib/raidfile/RaidFileController.h	2005-04-18 15:11:17.435141400 +0600
+++ boxbackup-0.09/lib/raidfile/RaidFileController.h	2005-04-18 17:40:20.070654160 +0600
@@ -120,7 +120,7 @@
 	~RaidFileController();
 	
 public:
-	void Initialise(const char *ConfigFilename = "/etc/box/raidfile.conf");
+	void Initialise(const char *ConfigFilename = "/etc/boxbackup/raidfile.conf");
 	int GetNumDiscSets() {return mSetList.size();}
 
 	// --------------------------------------------------------------------------
