diff -Naur boxbackup-0.09-orig/lib/common/BoxPlatform.h boxbackup-0.09/lib/common/BoxPlatform.h
--- boxbackup-0.09-orig/lib/common/BoxPlatform.h	2005-08-09 19:51:44.000000000 +0200
+++ boxbackup-0.09/lib/common/BoxPlatform.h	2005-08-09 19:51:57.000000000 +0200
@@ -117,7 +117,6 @@
 	#include <sys/types.h>
 	
 	// types 'missing'
-	typedef int socklen_t;
 	typedef u_int8_t uint8_t;
 	typedef signed char int8_t;
 	typedef u_int64_t uint64_t;
