--- src/kdar.desktop.o	2009-02-17 00:36:09.000000000 +0100
+++ src/kdar.desktop	2009-02-17 00:36:19.000000000 +0100
@@ -1,9 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=KDar (Disk-based Archive Tool)
 Exec=kdar
 Icon=kdar
 Type=Application
-DocPath=kdar/index.html
+X-DocPath=kdar/index.html
 Comment=A KDE frontend to DAR (Disk ARchive)
-Terminal=0
+Terminal=false
--- src/Makefile.am.o	2009-02-17 00:36:43.000000000 +0100
+++ src/Makefile.am	2009-02-17 00:36:58.000000000 +0100
@@ -52,7 +52,7 @@
 KDE_ICON = AUTO
 
 # this is where the kdelnk file will go 
-kdelnkdir   = $(kde_appsdir)/Utilities
+kdelnkdir   = $(xdg_appsdir)
 kdelnk_DATA = kdar.desktop
 
 # this is where the XML-GUI resource file goes
