diff -urN bacula-5.0.1.orig/autoconf/configure.in bacula-5.0.1/autoconf/configure.in
--- bacula-5.0.1.orig/autoconf/configure.in	2010-02-24 16:33:48.000000000 +0100
+++ bacula-5.0.1/autoconf/configure.in	2010-03-22 16:51:57.282119344 +0100
@@ -823,7 +823,7 @@
 if test x$support_conio = xyes; then
    AC_CHECK_HEADER(termcap.h, 
      [ AC_CHECK_LIB(termcap, tgetent, 
-       [ CONS_LIBS="-ltermcap"
+       [ CONS_LIBS="-lncurses"
     CONS_OBJ="conio.o"
     CONS_SRC="conio.c"
     got_conio="yes"
@@ -899,7 +899,7 @@
 	     AC_CHECK_HEADER(${with_readline}/readline.h, 
 		[
 		    AC_DEFINE(HAVE_READLINE, 1, [Set to enable readline support])
-		    CONS_LIBS="-lreadline -lhistory -ltermcap"
+		    CONS_LIBS="-lreadline -lhistory -lncurses"
 		    got_readline="yes"	 
 		], [
 		    echo " "
@@ -916,7 +916,7 @@
 		AC_DEFINE(HAVE_READLINE, 1, [Set to enable readline support])
 		got_readline="yes"
 		CONS_INC="-I/usr/include/readline"
-		CONS_LIBS="-lreadline -ltermcap"
+		CONS_LIBS="-lreadline -lncurses"
 	    ], [
 		dnl Did not find standard library, so try Bacula's default
 		AC_CHECK_HEADER(${TOP_DIR}/depkgs/readline/readline.h, 
@@ -924,7 +924,7 @@
 			AC_DEFINE(HAVE_READLINE, 1, [Set to enable readline support])
 			got_readline="yes"   
 			CONS_INC="-I${TOP_DIR}/depkgs/readline"
-			CONS_LIBS="-lreadline -lhistory -ltermcap"
+			CONS_LIBS="-lreadline -lhistory -lncurses"
 			CONS_LDFLAGS="-L${TOP_DIR}/depkgs/readline"
 			PRTREADLINE_SRC="${TOP_DIR}/depkgs/readline"
 		    ], [
